src/sentry/static/sentry/app/views/usageStats/index.tsx

```diff
@@ -1,63 +1,94 @@
 
+import {DateTimeObject} from 'app/components/charts/utils';
 import PageHeading from 'app/components/pageHeading';
+import {DEFAULT_RELATIVE_PERIODS, DEFAULT_STATS_PERIOD} from 'app/constants';
 import {t, tct} from 'app/locale';
 import {PageContent, PageHeader} from 'app/styles/organization';
-import {DataCategory, Organization} from 'app/types';
+import {DataCategory, Organization, RelativePeriod} from 'app/types';
 
-import {OrganizationUsageStats, ProjectUsageStats} from './types';
+import {ChartDataTransform} from './usageChart';
 import UsageStatsOrg from './usageStatsOrg';
-import UsageStatsProjects from './usageStatsProjects';
+// import UsageStatsProjects from './usageStatsProjects';
 
 type Props = {
   organization: Organization;
-  orgStatsLoading: boolean;
-  projectStatsLoading: boolean;
-  orgStats?: OrganizationUsageStats;
-  orgStatsError?: Error;
-  projectStats?: ProjectUsageStats[];
-  projectStatsError?: Error;
 } & RouteComponentProps<{orgId: string}, {}>;
 
-type State = {
-  dataCategory: DataCategory;
-  dateStart: moment.Moment;
-  dateEnd: moment.Moment;
-};
-
-class OrganizationStats extends React.Component<Props, State> {
-  state: State = {
-    dataCategory: DataCategory.ERRORS,
-    dateStart: moment().subtract(14, 'days'),
-    dateEnd: moment(),
-  };
+class OrganizationStats extends React.Component<Props> {
+  get dataCategory(): DataCategory {
+    const dataCategory = this.props.location?.query?.dataCategory;
 
-  setDataCategory = (dataCategory: DataCategory) => {
-    this.setState({dataCategory});
-  };
+    switch (dataCategory) {
+      case DataCategory.ERRORS:
+      case DataCategory.TRANSACTIONS:
+      case DataCategory.ATTACHMENTS:
+        return dataCategory as DataCategory;
+      default:
+        return DataCategory.ERRORS;
+    }
+  }
 
-  setDateRange = (dateStart: moment.Moment, dateEnd: moment.Moment) => {
-    this.setState({dateStart, dateEnd});
-  };
+  get dataCategoryName(): string {
+    const dataCategory = this.dataCategory;
 
-  get dataCategoryName() {
-    switch (this.state.dataCategory) {
+    switch (dataCategory) {
       case DataCategory.ERRORS:
         return t('Errors');
       case DataCategory.TRANSACTIONS:
         return t('Transactions');
       case DataCategory.ATTACHMENTS:
         return t('Attachments');
       default:
-        return t('Unknown Data Category');
+        return DataCategory.ERRORS;
+    }
+  }
+
+  get dataPeriod(): DateTimeObject {
+    const {statsPeriod, start, end} = this.props.location?.query ?? {};
+    if (!statsPeriod && !start && !end) {
+      return {period: DEFAULT_STATS_PERIOD};
     }
+
+    // Absolute date range is more specific than period
+    if (start && end) {
+      return {start, end};
+    }
+
+    const keys = Object.keys(DEFAULT_RELATIVE_PERIODS);
+    return statsPeriod && keys.includes(statsPeriod)
+      ? {period: statsPeriod}
+      : {period: DEFAULT_STATS_PERIOD};
```

Comment:
What happens in the case where `!statsPeriod`, `start`, && `!end`?

Reply:
That'll be an invalid case and it'll return `{period: DEFAULT_STATS_PERIOD}`